SMP-induced failure on 8.4.1

```
    [1 of 1] Compiling Data.String.Builder ( src/Data/String/Builder.hs, dist/build/Data/String/Builder.o )

    src/Data/String/Builder.hs:44:10: error:
        • No instance for (Semigroup Builder)
            arising from the superclasses of an instance declaration
        • In the instance declaration for ‘Monoid Builder’
       |
    44 | instance Monoid Builder where
       |          ^^^^^^^^^^^^^^
    builder for ‘/nix/store/idknvpwa8q6mrdm69p6pj3ck1mxk535m-stringbuilder-0.5.0.drv’ failed with exit code 1
```

--- a/src/Data/String/Builder.hs
+++ b/src/Data/String/Builder.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveFunctor, TypeSynonymInstances, FlexibleInstances, TypeFamilies #-}
+{-# LANGUAGE CPP, DeriveFunctor, TypeSynonymInstances, FlexibleInstances, TypeFamilies #-}
 -- |
 -- The `build` function can be used to construct multi-line string literals in
 -- a monadic way:
@@ -43,7 +43,12 @@ type Builder = BuilderM ()
 
 instance Monoid Builder where
   mempty = return ()
-  a `mappend` b = a >> b
+#if !MIN_VERSION_base(4,11,0)
+  mappend = (>>)
+#else
+instance Semigroup Builder where
+  (<>) = (>>)
+#endif
 
 -- | Add a literal string.
 literal :: String -> Builder

